README.md

```diff
@@ -7,3 +7,74 @@ This repository contains ES6 JS implementations of the Streamr [protocol](PROTOC
  - [broker](https://github.com/streamr-dev/broker)
  
  This package is available on npm as `streamr-client-protocol`.
+
+### Usage
+
+This section describes how to use the Javascript implementation of the [protocol](PROTOCOL.md).
+
+#### Creating messages from arguments
+
+Every message type from both the Control Layer and the Message Layer is defined as a class and has a static `create` method that takes class-specific arguments to build an instance of the latest version of the message type.
+
+This example shows how to build a `StreamMessage` and encapsulate it in a  `UnicastMessage`.
+
+```javascript
+const content = {
+                foo: 'bar',
+            }
+const streamMessage = StreamMessage.create(['streamId', 0, Date.now(), 0, 'publisherId', 'msgChainId'],
+                                             null, StreamMessage.CONTENT_TYPES.JSON, content,
+                                             StreamMessage.SIGNATURE_TYPES.NONE, null)
+const unicastMessage = UnicastMessage.create("subscriptionId", streamMessage)
+```
+
+#### Serializing messages to JSON arrays or strings
+
+Every message type from both the Control Layer and the Message Layer has a `serialize` method that takes different optional arguments depending on the message type. By default, it serializes to the latest version of the message type and the result is a string.
```

Comment:
All `serialize` methods take a `version` as the first argument, right? Worth mentioning here!